### PR TITLE
Update Docusaurus to 3.4.0 to fix relative `.md` handling

### PR DESCRIPTION
**Pull request by nuke-web3**
https://github.com/facebook/docusaurus/issues/10152 is the issue blocking Vercel so we must update for that to work correctly.

Targets #1972 branch

### CODE REVIEW

File: website/sidebars.js [hunk 1]
```diff
@@ -30,13 +30,18 @@ export default {
...
           type: "doc",
-          label: "Sequence Diagram",
-          id: "proof-system/proof-system-sequence-diagram",
+          label: "Execution Traces",
+          id: "proof-system/what-is-a-trace",
+        },
+        {
+          type: "doc",
+          label: "STARK by Hand",
+          id: "proof-system/stark-by-hand",
```
Comment (nuke-web3) [inline website/sidebars.js:44]: @pdg744 - not sure what order you want these in, but without them in the side bar we have errors, and those pages do not have a sidebar themselves

Comment (pdg744) [reply website/sidebars.js:44]: Not super opinionated about the ordering here. 

I guess my instinct is that the "What is an Execution Trace" doc may fit better in the `docs/reference-docs/` section along side "about-arithmetic-circuits" etc, but happy to go ahead with whatever feels simplest at the moment in order to get the link-checker working, and worry about perfecting the organization later.

**Review by pdg744 — APPROVED** [review]

**Comment by pdg744** [comment]
Sidenote: was surprised to see this PR seemed to indicate my involvement was necessary as code owner -- I assumed that would only be relevant for PRs that were targeting `main`.

**Review by flaub — APPROVED** [review]

**Comment by flaub** [comment]
Nice!

**Comment by flaub** [comment]
Our versioned docs only make sense on `main`, that is to say that older versions of the docs are still on the `main` branch since we only ever deploy from the `main` branch. (I hope that made sense!)

**Comment by pdg744** [comment]
> Our versioned docs only make sense on `main`, that is to say that older versions of the docs are still on the `main` branch since we only ever deploy from the `main` branch. (I hope that made sense!)

When I said "targeting main," I meant that this PR is just merging nuke's branch into the `flaub/lychee`. 
/shrug I guess code owner also applies to these kind of branch-to-branch PRs

**Pull request merged**